### PR TITLE
[SYCL] add deep copy for annotated_ref assignment operator

### PR DESCRIPTION
**Pull request by broxigarchen**
annotated_ref assignment should copy the underneath pointer

### CODE REVIEW

**Comment by broxigarchen** [comment]
@KseniyaTikhomirova  can you please help to review this PR? This is p1 for 2024.0 Thanks!

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc [hunk 1]
```diff
@@ -610,10 +610,11 @@ applying the annotations when the object is stored to memory.
 a|
 [source,c++]
 ----
-annotated_ref& operator=(const annotated_ref&) = default;
+annotated_ref& operator=(const annotated_ref&);
 ----
 |
-Assign from another `annotated_ref` object.
+Copy an object of type `T` from another `annotated_ref<T, ...>` object.
+applying the annotations when the object is loaded from and stored to memory.
```
Comment (KseniyaTikhomirova) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc:617]:
```suggestion
Applying the annotations when the object is loaded from and stored to memory.
```

**Review by KseniyaTikhomirova — APPROVED** [review]
LGTM

**Comment by broxigarchen** [comment]
@intel/llvm-gatekeepers can you help to take a look at this? Thanks!

**Review by steffenlarsen — APPROVED** [review]

**Pull request merged**